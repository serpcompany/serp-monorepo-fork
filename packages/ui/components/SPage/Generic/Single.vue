<script setup lang="ts">
  import type { Entity, Comment } from '@serp/types/types';

  const { user } = useUserSession();
  const route = useRoute();
  const router = useRouter();
  const slug = route.params.slug as string;

  const toast = useToast();

  const props = defineProps<{
    module: string;
    baseSlug: string;
    categoryBaseSlug: string;
  }>();

  const { module, baseSlug, categoryBaseSlug } = toRefs(props);

  const data = (await useEntity(slug, module.value)) as Entity;
  if (!data) {
    router.push('/404');
  }

  const isVerified = computed(() => {
    return data.verification === user.value?.siteId;
  });

  const config = useRuntimeConfig();
  const useAuth = config.public.useAuth;

  // Create a consolidated object with all provider details
  const providerDetails = computed(() => {
    return {
      basic: data.basicInfo || {},
      contracts: data.contracts || {},
      pricing: data.pricing || {},
      services: data.services || {},
      industries: data.industries || {},
      businessesServed: data.businessesServed || {},
      supportSetup: data.supportSetup || {},
      ratings: data.ratings || {}
    };
  });

  // @ts-expect-error: Auto-imported from another layer
  const { comments } = (await useEntityComments(data.id)) as {
    comments: Comment[];
  };

  // @ts-expect-error: Auto-imported from another layer
  const reviews = await useEntityReviews(data.id);
  reviews.entityId = data.id;

  // State for review modal
  const showReviewModal = ref(false);

  // Handle review submission - refresh reviews data
  async function handleReviewSubmitted() {
    // @ts-expect-error: Auto-imported from another layer
    const updatedReviews = await useEntityReviews(data.id);
    Object.assign(reviews, updatedReviews);
    reviews.entityId = data.id;
  }

  async function copyToClipboard(sectionId: string) {
    const jumpLink = `#${sectionId}`;
    const link = `${window.location.href.split('#')[0]}${jumpLink}`;

    try {
      if (navigator.clipboard && navigator.clipboard.writeText) {
        await navigator.clipboard.writeText(link);
      } else {
        // Fallback for unsupported environments
        const el = document.createElement('textarea');
        el.value = link;
        document.body.appendChild(el);
        el.select();
        document.execCommand('copy');
        document.body.removeChild(el);
      }

      toast.add({
        id: 'copy-link',
        title: 'Link copied to clipboard',
        description: `Link to ${sectionId} section copied successfully`,
        icon: 'check-circle'
      });
    } catch (error) {
      console.error('Failed to copy text to clipboard:', error);
      toast.add({
        id: 'copy-link-error',
        title: 'Failed to copy link',
        description: `Could not copy link to ${sectionId} section`,
        icon: 'error-circle'
      });
    }
  }

  useSeoMeta({
    title: computed(() => (data.name ? `${data.name}` : 'Unknown'))
  });
</script>

<template>
  <UPage v-if="data">
    <MultipageHeader
      :name="data.name"
      :one-liner="data.oneLiner || data.description"
      :sections="sections"
      class="bg-background sticky top-0 z-50 transition-all duration-300"
      :image="data.image || data.logo || data.logoUrl"
      :serply-link="data.serplyLink"
    >
      <template #upvote>
        <EditButton v-if="useAuth" :id="data.id" />
        <VerificationButton
          v-if="useAuth"
          :id="data.id"
          :domain="data.slug"
          :is-verified-prop="data.verification"
        />
        <VoteButton
          v-if="useAuth"
          :id="data.id"
          :module="module"
          :users-current-vote="data.usersCurrentVote"
          :upvotes="data.numUpvotes"
          :downvotes="data.numDownvotes"
        />
      </template>
    </MultipageHeader>

    <!-- Main content - single column layout -->
    <section class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
      <JSONRenderer
        v-if="data"
        :value="data"
        :base-slug="baseSlug"
        :category-base-slug="categoryBaseSlug"
      />
    </section>

    <UCard
      id="reviews"
      class="mb-8 scroll-mt-60 rounded-md border border-gray-200 dark:border-gray-800"
      :ui="{ body: { padding: 'p-0' } }"
    >
      <template #header>
        <div class="flex items-center px-4 pt-4 pb-2 sm:px-6 sm:pt-6">
          <h2 class="text-xl font-semibold text-gray-900 dark:text-gray-100">
            {{ data.name }} Reviews
          </h2>
          <UIcon
            name="i-heroicons-link"
            class="ml-2 h-4 w-4 text-gray-400 hover:cursor-pointer"
            @click="copyToClipboard('reviews')"
          />
        </div>
      </template>
      <UDivider class="my-0" />
      <div class="py-4 sm:p-6">
        <ReviewDistributionCard
          :total-reviews="data.numReviews ? data.numReviews : 0"
          :total-one-star-reviews="
            data.numOneStarReviews ? data.numOneStarReviews : 0
          "
          :total-two-star-reviews="
            data.numTwoStarReviews ? data.numTwoStarReviews : 0
          "
          :total-three-star-reviews="
            data.numThreeStarReviews ? data.numThreeStarReviews : 0
          "
          :total-four-star-reviews="
            data.numFourStarReviews ? data.numFourStarReviews : 0
          "
          :total-five-star-reviews="
            data.numFiveStarReviews ? data.numFiveStarReviews : 0
          "
          :average-rating="data.averageRating ? data.averageRating : 0"
          :show-border="false"
          card-title=""
          class="pb-12"
          :show-review-button="useAuth"
          @open-review-form="showReviewModal = true"
        />

        <!-- Display Reviews List -->
        <Reviews :is-verified="isVerified" :reviews="reviews" class="mt-8" />
      </div>
    </UCard>

    <ReviewModal
      v-model:open="showReviewModal"
      :entity-id="data.id"
      :result="reviews"
      @close="showReviewModal = false"
      @review-submitted="handleReviewSubmitted"
    />

    <!-- Discussion Section -->
    <UCard
      id="discussion"
      class="mb-8 scroll-mt-30 rounded-md border border-gray-200 dark:border-gray-800"
      :ui="{ body: { padding: 'p-0' } }"
    >
      <template #header>
        <div class="flex items-center px-4 pt-4 pb-2 sm:px-6 sm:pt-6">
          <h2 class="text-xl font-semibold text-gray-900 dark:text-gray-100">
            Discussion
          </h2>
          <UIcon
            name="i-heroicons-link"
            class="ml-2 h-4 w-4 text-gray-400 hover:cursor-pointer"
            @click="copyToClipboard('discussion')"
          />
        </div>
      </template>
      <UDivider class="my-0" />
      <div class="p-4 sm:p-6">
        <CommentsContainer
          v-if="useAuth"
          :id="data.id"
          :module="data.module"
          :comments="comments"
          class="comments-github-style"
        />
      </div>
    </UCard>
  </UPage>
</template>
