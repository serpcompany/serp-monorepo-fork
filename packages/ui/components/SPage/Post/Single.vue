<script setup lang="ts">
  const props = defineProps<{ module?: string }>();

  const computedModule = computed(() => props.module);

  const route = useRoute();
  const slug = route.params.slug as string;
  const router = useRouter();
  const data = await usePost(slug, computedModule.value);
  if (!data) {
    router.push('/404');
  }

  useSeoMeta({
    title: () => data.title || data.name
  });
</script>

<template>
  <UMain>
    <SinglePostsGlossaryPost v-if="data.module === 'Glossary'" :data="data" />
    <SinglePostsPost v-else :data="data" :module="computedModule || ''" />
  </UMain>
</template>
