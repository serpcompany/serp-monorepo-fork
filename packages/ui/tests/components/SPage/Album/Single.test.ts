import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import Single from '../../../../components/SPage/Album/Single.vue';
import ComponentRender from '../../../componentRender';
import '../../../mockUseUserSession';

let config_: Record<string, unknown> = {
  app: { baseURL: '/' },
  public: { useAuth: true }
};
let albumData_: unknown = {};
let upvotesData_: unknown = { upvotes: [] };

mockNuxtImport('useRuntimeConfig', () => () => config_);
mockNuxtImport('useSeoMeta', () => () => {});
mockNuxtImport('useAlbum', () => () => Promise.resolve(albumData_));
mockNuxtImport('useFetchWithCache', () => () => Promise.resolve(upvotesData_));

const scenarios: [
  string,
  { album: Record<string, unknown>; upvotes: string[]; useAuth: boolean }
][] = [
  [
    'with auth enabled and full album data',
    {
      album: {
        name: 'Test Album',
        slug: 'test-album',
        overview: 'This is a test album overview.',
        recordings: [
          { slug: 'song-one', name: 'Song One', position: 1, has_lyrics: true },
          { slug: 'song-two', name: 'Song Two', position: 2, has_lyrics: false }
        ],
        coverArt: { '500': '/test-cover.jpg' },
        date: '2020-01-01',
        genres: ['Rock', 'Pop'],
        tags: ['Indie', 'Live'],
        id: 'album123',
        seoTitle: 'Test Album SEO Title',
        seoDescription: 'Test Album SEO Description'
      },
      upvotes: ['user@example.com'],
      useAuth: true
    }
  ],
  [
    'with auth disabled and minimal album data',
    {
      album: {
        name: 'Minimal Album',
        slug: 'minimal-album',
        overview: 'Minimal album overview.',
        recordings: [],
        coverArt: {},
        date: null,
        genres: [],
        tags: [],
        id: 'album456',
        seoTitle: '',
        seoDescription: ''
      },
      upvotes: [],
      useAuth: false
    }
  ],
  [
    'with auth enabled and no recordings',
    {
      album: {
        name: 'No Songs Album',
        slug: 'no-songs-album',
        overview: 'Overview of album with no songs.',
        recordings: [],
        coverArt: { '500': '/no-songs-cover.jpg' },
        date: '2021-06-15',
        genres: ['Jazz'],
        tags: ['Smooth'],
        id: 'album789',
        seoTitle: 'No Songs SEO',
        seoDescription: 'No Songs Description'
      },
      upvotes: ['anotheruser@example.com'],
      useAuth: true
    }
  ]
];

describe('SPageAlbumSingle Snapshot', () => {
  it.each(scenarios)(
    'renders %s correctly',
    async (desc, { album, upvotes, useAuth }) => {
      albumData_ = album;
      upvotesData_ = { upvotes };
      config_ = {
        app: { baseURL: '/' },
        public: { useAuth }
      };

      const html = await ComponentRender(
        `SPageAlbumSingle ${desc}`,
        {},
        Single
      );
      expect(html).toMatchSnapshot();
    }
  );
});
