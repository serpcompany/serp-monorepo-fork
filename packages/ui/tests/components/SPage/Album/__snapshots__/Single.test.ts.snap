// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageAlbumSingle Snapshot > renders with auth disabled and minimal album data correctly 1`] = `
"<div>
  <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
    <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
        <div class="flex min-w-0 flex-grow items-center gap-3">
          <!--v-if-->
          <div class="min-w-0 flex-grow">
            <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">Minimal Album</h1>
            <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a></div>
      </div>
      <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#overview" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Overview</span>
          <!--v-if-->
        </a><a href="#songs" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Songs</span>
          <!--v-if-->
        </a></nav>
    </div>
  </header><!-- Main content wrapper -->
  <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
    <!-- Breadcrumbs -->
    <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
      <ol class="flex items-center gap-1.5"></ol>
    </nav><!-- Grid Layout -->
    <div class="grid gap-6 lg:grid-cols-3">
      <!-- Main Content Column (70%) -->
      <div class="space-y-8 lg:col-span-2">
        <!-- Songs / Tracklist Section -->
        <section id="songs" class="scroll-mt-24" data-adsense-content="album-tracklist">
          <h2 class="mb-4 text-2xl font-semibold">Tracklist</h2>
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <!--v-if-->
            <div class="p-4 sm:p-6 [object Object]">
              <p class="text-sm text-gray-500 dark:text-gray-400"> No tracklist available. </p>
            </div>
            <!--v-if-->
          </div>
        </section><!-- Overview Section -->
        <section id="overview" class="scroll-mt-24" data-adsense-content="album-overview">
          <h2 class="mb-4 text-2xl font-semibold">Overview</h2>
          <div class="prose dark:prose-invert max-w-none">Minimal album overview.</div>
        </section><!-- Other Albums by Artist Section -->
        <!--v-if-->
      </div><!-- Sidebar Column (30%) -->
      <div class="space-y-6 lg:col-span-1">
        <!-- Album image -->
        <!--v-if-->
        <!-- Album Info Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Album Info</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <!-- Artist Link -->
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
        </div><!-- Stats Card -->
        <!--v-if-->
        <!-- Links Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Listen On</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                <!--v-if--> Amazon Music
                <!--v-if-->
              </a></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`SPageAlbumSingle Snapshot > renders with auth enabled and full album data correctly 1`] = `
"<div>
  <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
    <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
        <div class="flex min-w-0 flex-grow items-center gap-3">
          <!--v-if-->
          <div class="min-w-0 flex-grow">
            <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">Test Album</h1>
            <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto">
          <div module="music_albums">
            <div class="relative inline-flex -space-x-px"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-5" aria-hidden="true"></span>
                <!--v-if-->
                <!--v-if-->
              </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1 rounded-md ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center"><!--v-if--><span class="truncate">NaN</span>
              <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-5" aria-hidden="true"></span>
                <!--v-if-->
                <!--v-if-->
              </button>
            </div>
          </div><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a>
        </div>
      </div>
      <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#overview" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Overview</span>
          <!--v-if-->
        </a><a href="#songs" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Songs</span>
          <!--v-if-->
        </a></nav>
    </div>
  </header><!-- Main content wrapper -->
  <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
    <!-- Breadcrumbs -->
    <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
      <ol class="flex items-center gap-1.5"></ol>
    </nav><!-- Grid Layout -->
    <div class="grid gap-6 lg:grid-cols-3">
      <!-- Main Content Column (70%) -->
      <div class="space-y-8 lg:col-span-2">
        <!-- Songs / Tracklist Section -->
        <section id="songs" class="scroll-mt-24" data-adsense-content="album-tracklist">
          <h2 class="mb-4 text-2xl font-semibold">Tracklist</h2>
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <!--v-if-->
            <div class="p-4 sm:p-6 [object Object]">
              <ol class="space-y-2">
                <li class="flex items-center justify-between gap-2 border-b border-gray-200 pb-2 last:border-b-0 last:pb-0 dark:border-gray-700">
                  <div class="flex min-w-0 items-center gap-3"><span class="w-5 text-right font-mono text-xs text-gray-500 dark:text-gray-400">1.</span>
                    <div class="truncate"><a href="/songs/song-one" class="text-primary-600 dark:text-primary-400 truncate text-sm font-medium hover:underline" title="Song One">Song One</a></div>
                  </div><span class="flex-shrink-0 font-mono text-xs text-gray-500 dark:text-gray-400">--:--</span>
                </li>
                <li class="flex items-center justify-between gap-2 border-b border-gray-200 pb-2 last:border-b-0 last:pb-0 dark:border-gray-700">
                  <div class="flex min-w-0 items-center gap-3"><span class="w-5 text-right font-mono text-xs text-gray-500 dark:text-gray-400">2.</span>
                    <div class="truncate"><span class="truncate text-sm text-gray-800 dark:text-gray-200" title="Song Two">Song Two</span></div>
                  </div><span class="flex-shrink-0 font-mono text-xs text-gray-500 dark:text-gray-400">--:--</span>
                </li>
              </ol>
            </div>
            <!--v-if-->
          </div>
        </section><!-- Overview Section -->
        <section id="overview" class="scroll-mt-24" data-adsense-content="album-overview">
          <h2 class="mb-4 text-2xl font-semibold">Overview</h2>
          <div class="prose dark:prose-invert max-w-none">This is a test album overview.</div>
        </section><!-- Other Albums by Artist Section -->
        <!--v-if-->
      </div><!-- Sidebar Column (30%) -->
      <div class="space-y-6 lg:col-span-1">
        <!-- Album image -->
        <div><img alt="Test Album cover art" data-nuxt-img="" srcset="/_ipx/_/test-cover.jpg 1x, /_ipx/_/test-cover.jpg 2x" class="w-full rounded-md object-cover shadow-md" src="/_ipx/_/test-cover.jpg"></div><!-- Album Info Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Album Info</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <!-- Artist Link -->
              <!--v-if-->
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Released: </span><span class="text-gray-800 dark:text-gray-200">2020</span></div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
        </div><!-- Stats Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Stats</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Genres: </span><span class="text-gray-800 dark:text-gray-200">Rock, Pop</span></div>
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Tags: </span><span class="text-gray-800 dark:text-gray-200">Indie, Live</span></div>
            </div>
          </div>
          <!--v-if-->
        </div><!-- Links Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Listen On</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                <!--v-if--> Amazon Music
                <!--v-if-->
              </a></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`SPageAlbumSingle Snapshot > renders with auth enabled and no recordings correctly 1`] = `
"<div>
  <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
    <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
        <div class="flex min-w-0 flex-grow items-center gap-3">
          <!--v-if-->
          <div class="min-w-0 flex-grow">
            <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">No Songs Album</h1>
            <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
          </div>
        </div>
        <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto">
          <div module="music_albums">
            <div class="relative inline-flex -space-x-px"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-5" aria-hidden="true"></span>
                <!--v-if-->
                <!--v-if-->
              </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1 rounded-md ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center"><!--v-if--><span class="truncate">NaN</span>
              <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-5" aria-hidden="true"></span>
                <!--v-if-->
                <!--v-if-->
              </button>
            </div>
          </div><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
            <!--v-if-->
          </a>
        </div>
      </div>
      <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#overview" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Overview</span>
          <!--v-if-->
        </a><a href="#songs" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
          <!--v-if--><span class="truncate">Songs</span>
          <!--v-if-->
        </a></nav>
    </div>
  </header><!-- Main content wrapper -->
  <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
    <!-- Breadcrumbs -->
    <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
      <ol class="flex items-center gap-1.5"></ol>
    </nav><!-- Grid Layout -->
    <div class="grid gap-6 lg:grid-cols-3">
      <!-- Main Content Column (70%) -->
      <div class="space-y-8 lg:col-span-2">
        <!-- Songs / Tracklist Section -->
        <section id="songs" class="scroll-mt-24" data-adsense-content="album-tracklist">
          <h2 class="mb-4 text-2xl font-semibold">Tracklist</h2>
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <!--v-if-->
            <div class="p-4 sm:p-6 [object Object]">
              <p class="text-sm text-gray-500 dark:text-gray-400"> No tracklist available. </p>
            </div>
            <!--v-if-->
          </div>
        </section><!-- Overview Section -->
        <section id="overview" class="scroll-mt-24" data-adsense-content="album-overview">
          <h2 class="mb-4 text-2xl font-semibold">Overview</h2>
          <div class="prose dark:prose-invert max-w-none">Overview of album with no songs.</div>
        </section><!-- Other Albums by Artist Section -->
        <!--v-if-->
      </div><!-- Sidebar Column (30%) -->
      <div class="space-y-6 lg:col-span-1">
        <!-- Album image -->
        <div><img alt="No Songs Album cover art" data-nuxt-img="" srcset="/_ipx/_/no-songs-cover.jpg 1x, /_ipx/_/no-songs-cover.jpg 2x" class="w-full rounded-md object-cover shadow-md" src="/_ipx/_/no-songs-cover.jpg"></div><!-- Album Info Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Album Info</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <!-- Artist Link -->
              <!--v-if-->
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Released: </span><span class="text-gray-800 dark:text-gray-200">2021</span></div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
        </div><!-- Stats Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Stats</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="space-y-2 text-sm">
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Genres: </span><span class="text-gray-800 dark:text-gray-200">Jazz</span></div>
              <div><span class="font-medium text-gray-700 dark:text-gray-300">Tags: </span><span class="text-gray-800 dark:text-gray-200">Smooth</span></div>
            </div>
          </div>
          <!--v-if-->
        </div><!-- Links Card -->
        <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
          <div class="p-4 sm:px-6">
            <div class="font-medium">Listen On</div>
          </div>
          <div class="p-4 sm:p-6">
            <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                <!--v-if--> Amazon Music
                <!--v-if-->
              </a></div>
          </div>
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
</div>"
`;
